CorsixTH/Lua/entities/object.lua

```diff
@@ -904,3 +904,35 @@ function Object:setState(state)
     self.times_used = state.times_used
   end
 end
+
+--! An object that lives on the edge of a tile.
+class "SideObject" (Object)
+
+---@type SideObject
+local SideObject = _G["SideObject"]
+
+function SideObject:SideObject(...)
+  self:Object(...)
+end
+
+function SideObject:getDrawingLayer()
+  if self.direction == "north" then
+    return DrawingLayers.NorthSideObject
+  elseif self.direction == "west" then
+    return DrawingLayers.WestSideObject
+  else
+    if self.direction == "east" then
+      if self.object_type.thob == 50 then
+      --[[ Some side objects, such as the bin, have two orientations north and east
+        but they are displayed in the north and west part of the tile respectively.
+        This could lead to a graphical glitch where a side object in the west part
+        of the tile is displayed over a doctor in the middle of the tile. ]]
+        return DrawingLayers.WestSideObject
```

Comment:
This case is strictly only the bin; not sure why they used the thob. So this comment was right before.